### PR TITLE
ARM OSX Migrator 

### PR DESCRIPTION
**Pull request by xiki-tempula**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by douglowe** [comment]
@conda-forge-admin, please rerender

**Comment by douglowe** [comment]
Our build problem is:
```
Traceback (most recent call last):
  File "/opt/conda/bin/conda-mambabuild", line 10, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.10/site-packages/boa/cli/mambabuild.py", line 256, in main
    call_conda_build(action, config)
  File "/opt/conda/lib/python3.10/site-packages/boa/cli/mambabuild.py", line 228, in call_conda_build
    result = api.build(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/api.py", line 253, in build
    return build_tree(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 3804, in build_tree
    packages_from_this = build(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 2673, in build
    utils.check_call_env(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/utils.py", line 445, in check_call_env
    return _func_defaulting_env_to_os_environ("call", *popenargs, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/conda_build/utils.py", line 421, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'errexit', '/home/conda/feedstock_root/build_artifacts/gromacs_1701103828945/work/conda_build.sh']' returned non-zero exit status 2.

```

File: .scripts/build_steps.sh [hunk 1]
```diff
@@ -71,7 +74,7 @@ if [[ "${BUILD_WITH_CONDA_DEBUG:-0}" == 1 ]]; then
     # Drop into an interactive shell
     /bin/bash
 else
-    conda build "${RECIPE_ROOT}" -m "${CI_SUPPORT}/${CONFIG}.yaml" \
+    conda mambabuild "${RECIPE_ROOT}" -m "${CI_SUPPORT}/${CONFIG}.yaml" \
```
Comment (douglowe) [inline .scripts/build_steps.sh:77]: I suggest reverting this change, and see if the packages will build with conda. If they do, then we are going to have to ditch mamba for the moment.

**Comment by douglowe** [comment]
But, on the upside - it looks like the ARM64 openmpi builds are no longer running.

**Review by jan-janssen — APPROVED** [review]
Looks good to me

**Comment by douglowe** [comment]
This is all looking good to me too.

However, please can you bump the build number (meta.yaml, line 6) before this is merged?

**Comment by xiki-tempula** [comment]
Sorry for the confusion. We had this pipeline problem so I just revert to the initial commit where the pipeline passes and then go one by one trying to sort things out.

**Comment by xiki-tempula** [comment]
Ok, I guess the reason that build.sh is missing an `fi` weird that it passed before.

**Comment by xiki-tempula** [comment]
@conda-forge-admin, please rerender

**Comment by xiki-tempula** [comment]
@conda-forge-admin, please rerender

**Comment by xiki-tempula** [comment]
Ok, I have finally found out why the pipeline fails. I kind of based my branch on conda-forge/gromacs-feedstock:bot-pr_arm_osx_h742397 and assume that it is the same branch as the one in https://github.com/conda-forge/gromacs-feedstock/pull/30, but they are not the same branch and missed two crucial commits which is why my pipeline is always failing.

**Comment by xiki-tempula** [comment]
@douglowe I think it is working now. Do you mind merge this PR, please? Thank you.

File: recipe/build.sh [hunk 2]
```diff
@@ -3,8 +3,22 @@
 mkdir build
 cd build
 
+# Temporary debugging output checking how we're doing cross-compiling
+uname
+uname -m
+echo $HOST
+
```
Comment (mabraham) [inline recipe/build.sh:10]:
```suggestion
```

**Pull request merged**